src/Sylius/Bundle/OrderBundle/NumberGenerator/SequentialOrderNumberGenerator.php

```diff
@@ -64,11 +76,30 @@ public function __construct(
      */
     public function generate(OrderInterface $order)
     {
-        $sequence = $this->getSequence();
-        
-        $number = $this->generateNumber($sequence->getIndex());
-        $sequence->incrementIndex();
-        
+        $number = null;
+        $counter = 0;
+        $done = false;
+        while (!$done) {
+            if (++$counter > 100) {
```

Comment:
Maybe we should pass 100 as a argument to constructor? Or add a constant for that?